### PR TITLE
CBDC-1099 [DevOps] Sora plugin JDK 21

### PR DESCRIPTION
**Pull request by marat-biriushev**
# Task

[CBDC-1099] Sora plugin JDK 21

## Changes

1. Bumped gradle version to 8.5.
2. Added default image for java version 21.
3. Updated deprecated actions/checkout@v2.

## Author

Signed-off-by: Marat Biryushev vfhfnvvv@gmail.com

[CBDC-1099]: https://soramitsu.atlassian.net/browse/CBDC-1099

### CODE REVIEW

File: src/main/groovy/jp/co/soramitsu/devops/docker/DockerPlugin.groovy [hunk 1]
```diff
@@ -293,9 +295,11 @@ class DockerPlugin implements Plugin<Project> {
             return 'openjdk:13'
         } else if (javaVersion == 17) {
             return 'openjdk:17-jdk-slim'
+        } else if (javaVersion == 21) {
+            return 'amazoncorretto:21-alpine3.18-jdk'
```
Comment (Mingela) [inline src/main/groovy/jp/co/soramitsu/devops/docker/DockerPlugin.groovy:299]: why not remain on the same provider as older versions? e.g. `openjdk:21-jdk-slim`

File: src/main/groovy/jp/co/soramitsu/devops/docker/DockerPlugin.groovy [hunk 2]
```diff
@@ -293,9 +295,11 @@ class DockerPlugin implements Plugin<Project> {
...
         } else {
             // default fallback version
             return 'openjdk:8-jre-alpine'
         }
     }
-}
```
Comment (Mingela) [inline src/main/groovy/jp/co/soramitsu/devops/docker/DockerPlugin.groovy:301]: I guess this change should be reverted (empty last line to be present)

**Review by Mingela — COMMENTED** [review]
consider minor comments above

File: src/main/groovy/jp/co/soramitsu/devops/docker/DockerPlugin.groovy (hunk 1 above)
Comment (bbr88) [reply src/main/groovy/jp/co/soramitsu/devops/docker/DockerPlugin.groovy]: The corretto images are way better maintained. 0 known vulnerabilities while the openjdk has quite a few issues. 

Screenshot 2024-04-03 at 11 30 07
Screenshot 2024-04-03 at 11 30 35
Screenshot 2024-04-03 at 11 37 07

File: .github/workflows/github-actions.yml [hunk 3]
```diff
@@ -5,7 +5,7 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       # Checks-out the repo under $GITHUB_WORKSPACE, so that workflow accesses it
-      - uses: actions/checkout@v2
+      - uses: actions/checkout@v4
```
Comment (wpi86) [inline .github/workflows/github-actions.yml:8]: pin version

File: src/main/groovy/jp/co/soramitsu/devops/docker/DockerPlugin.groovy [hunk 4]
```diff
@@ -293,6 +295,8 @@ class DockerPlugin implements Plugin<Project> {
             return 'openjdk:13'
         } else if (javaVersion == 17) {
             return 'openjdk:17-jdk-slim'
+        } else if (javaVersion == 21) {
+            return 'openjdk:21-jdk-slim'
```
Comment (wpi86) [inline src/main/groovy/jp/co/soramitsu/devops/docker/DockerPlugin.groovy:299]: amazon images?

**Review by wpi86 — CHANGES_REQUESTED** [review]
did you test it with Bisarnov?

File: src/main/groovy/jp/co/soramitsu/devops/docker/DockerPlugin.groovy (hunk 1 above)
Comment (Mingela) [reply src/main/groovy/jp/co/soramitsu/devops/docker/DockerPlugin.groovy]: I realize that, true, though those are default backup images and they're still widely used. In any corporate project a maintainer is intended to follow DevOps recommendations and use other suitable image

File: src/main/groovy/jp/co/soramitsu/devops/docker/DockerPlugin.groovy (hunk 4 above)
Comment (Mingela) [reply src/main/groovy/jp/co/soramitsu/devops/docker/DockerPlugin.groovy:299]: I'd suggest unusual images to be set on the project level, not in the plugin

**Review by Mingela — APPROVED** [review]

File: .github/workflows/github-actions.yml (hunk 3 above)
Comment (marat-biriushev) [reply .github/workflows/github-actions.yml]: Pinned.

**Review by bbr88 — APPROVED** [review]

File: src/main/groovy/jp/co/soramitsu/devops/docker/DockerPlugin.groovy (hunk 4 above)
Comment (wpi86) [reply src/main/groovy/jp/co/soramitsu/devops/docker/DockerPlugin.groovy:299]: why not jre?

**Review by wpi86 — CHANGES_REQUESTED** [review]

File: src/main/groovy/jp/co/soramitsu/devops/docker/DockerPlugin.groovy (hunk 1 above)
Comment (marat-biriushev) [reply src/main/groovy/jp/co/soramitsu/devops/docker/DockerPlugin.groovy]: This [image](https://hub.docker.com/_/openjdk) is officially deprecated and all users are recommended to find and use suitable replacements ASAP.
eclipse-temurin:21-jre image tested.

File: src/main/groovy/jp/co/soramitsu/devops/docker/DockerPlugin.groovy (hunk 4 above)
Comment (marat-biriushev) [reply src/main/groovy/jp/co/soramitsu/devops/docker/DockerPlugin.groovy]: Tested:
eclipse-temurin:17-jre-alpine,
eclipse-temurin:21-jre-alpine.

**Review by wpi86 — APPROVED** [review]

**Pull request merged**